rnd/autogpt_server/autogpt_server/data/block.py

```diff
@@ -122,6 +125,46 @@ def get_required_fields(cls) -> set[str]:
             if field_info.is_required()
         }
 
+    @classmethod
+    def __pydantic_init_subclass__(cls, **kwargs):
+        """Validates the schema definition. Rules:
+        - Only one `CredentialsMetaInput` field may be present.
+          - This field MUST be called `credentials`.
+        - A field that is called `credentials` MUST be a `CredentialsMetaInput`.
```

Comment:
Why so strict, what edge cases you think of can be problematic?
Maybe just make a generic `class CredentialsBlock(Block)` or `class Credentials` (used like: `ConcreteBlockWithCredentials(Block, Credentials)`) that has `credentials` field.